challenge-two/src/agent.spec.ts

```diff
@@ -2,204 +2,208 @@ import { FindingType, FindingSeverity, HandleTransaction } from "forta-agent";
-    amountInMaximum: BigNumber.from(1980),
-    sqrtPriceLimitX96: BigNumber.from(0),
+  const TEST_VAL2 = {
+    TOKEN2_ADDR: createAddress("0x04"),
+    TOKEN2_VAL: ethers.BigNumber.from("100"),
+    TOKEN3_ADDR: createAddress("0x05"),
+    TOKEN3_VAL: ethers.BigNumber.from("400"),
+    POOL_ADDR2: createAddress("0x06"),
+    Fee: ethers.BigNumber.from("500"),
   };
 
   beforeAll(() => {
-    handleTransaction = provideHandleTransaction(
-      UNISWAP_ROUTER_ABI,
-      mockSwapRouterAddress
+    mockProvider = new ethers.providers.JsonRpcProvider(
+      "http://localhost:8545"
     );
+    handleTransaction = provideHandleTransaction(mockProvider);
   });
 
   beforeEach(() => {
-    mockTxEvent = new TestTransactionEvent();
+    jest.resetAllMocks();
   });
 
   it("returns empty findings if there are no swap events", async () => {
-    const findings = await handleTransaction(mockTxEvent);
+    const txEvent = new TestTransactionEvent();
+    const findings = await handleTransaction(txEvent);
     expect(findings).toStrictEqual([]);
   });
 
-  it("returns correct findings if there is one exactInputSingle swap event", async () => {
-    mockTxEvent = new TestTransactionEvent()
-      .setTo(mockSwapRouterAddress)
-      .setData(
-        SWAP_ABI.encodeFunctionData("exactInputSingle", [
-          mockExactInputSingleParams,
-        ])
-      );
-
-    const findings = await handleTransaction(mockTxEvent);
-
-    expect(findings).toStrictEqual([
-      expect.objectContaining({
-        alertId: "FORTA-2",
-        name: "Nethermind Forta Bot UniSwap - exactInputSingle",
-        description: `Swap event detected on ${mockSwapRouterAddress}`,
-        severity: FindingSeverity.Low,
-        type: FindingType.Info,
-        metadata: {
-          tokenIn: mockExactInputSingleParams.tokenIn,
-          tokenOut: mockExactInputSingleParams.tokenOut,
-          fee: mockExactInputSingleParams.fee.toString(),
-          recipient: mockExactInputSingleParams.recipient,
-          amountIn: mockExactInputSingleParams.amountIn.toString(),
-          amountOutMinimum:
-            mockExactInputSingleParams.amountOutMinimum.toString(),
-          sqrtPriceLimitX96:
-            mockExactInputSingleParams.sqrtPriceLimitX96.toString(),
-        },
-      }),
-    ]);
-  });
-
-  it("returns correct findings if there is one exactOutputSingle swap event", async () => {
-    mockTxEvent = new TestTransactionEvent()
-      .setTo(mockSwapRouterAddress)
-      .setData(
-        SWAP_ABI.encodeFunctionData("exactOutputSingle", [
-          mockExactOutputSingleParams,
-        ])
-      );
-
-    const findings = await handleTransaction(mockTxEvent);
-
+  it("returns correct findings if there is a valid swap event from a Uniswap V3 pool", async () => {
+    // Create a mock contract instance for the pool
+    const mockPoolContract = {
+      callStatic: {
+        factory: jest.fn().mockResolvedValue(UNISWAP_V3_FACTORY_ADDRESS),
+      },
+    };
```

Comment:
In the same way that this `mockPoolContract` simply returns the UniV3 Factory address when its `factory` getter is called, any contract deployed can do so.